### ISSUE TITLE
[AOTInductor] Need support to export freezing model

### ISSUE THREAD

**Issue by imzhuhl** [pytorch/pytorch#114450]
### 🚀 The feature, motivation and pitch

AOTInductor not support freezing because can not deal with opaque tensor.
I've come across a need to deploy a quantized model, which means I need to combine aoit and pt2e quant. The first obstacle encountered is inductor freezing. Open inductor config freezing before aot_compile will see the error: "NotImplementedError: Cannot access storage of OpaqueTensorImpl".

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

**Comment by leslie-fang-intel** [comment]
@imzhuhl thanks for reporting this issue. I think @chunyuan-w is already working into the fix. cc @chunyuan-w @jgong5 for more details.

**Comment by chunyuan-w** [comment]
Hi @imzhuhl, the freezing support in AOTInductor on CPU is not ready yet and we're still working on it.

**Comment by hl475** [comment]
@chunyuan-w - wondering if there is any update on this?

**Comment by chunyuan-w** [comment]
> @chunyuan-w - wondering if there is any update on this?

Hi @hl475, https://github.com/pytorch/pytorch/pull/114451#discussion_r1411521635 mentions that this feature requires the serialization and the deserialization support for MKLDNN tensors, which needs the corresponding support in oneDNN. oneDNN team is working on an internal PR to add API for this support. Once this API from oneDNN is ready and added into a release version, we'll submit a PR to upgrade oneDNN in PyTorch to include this new API and another PR to add the integration in inductor to support the current feature.

**Comment by chunyuan-w** [comment]
> > @chunyuan-w - wondering if there is any update on this?
> 
> Hi @hl475, [#114451 (comment)](https://github.com/pytorch/pytorch/pull/114451#discussion_r1411521635) mentions that this feature requires the serialization and the deserialization support for MKLDNN tensors, which needs the corresponding support in oneDNN. oneDNN team is working on an internal PR to add API for this support. Once this API from oneDNN is ready and added into a release version, we'll submit a PR to upgrade oneDNN in PyTorch to include this new API and another PR to add the integration in inductor to support the current feature.

oneDNN has added the serialization and the deserialization support of the memory descriptor in the [v3.4.1 release](https://github.com/oneapi-src/oneDNN/releases/tag/v3.4.1). We created draft PR to upgrade oneDNN to v3.4.1 in PyTorch and we're running the validation for this upgrade: https://github.com/pytorch/pytorch/pull/122472. AOTInductor related support of freezing model will be submitted when the upgrade PR landed.

**Comment by chunyuan-w** [comment]
Hi @imzhuhl, the support of freezing https://github.com/pytorch/pytorch/pull/124350 has landed in to the main branch. Let us know if you meet any other issues. And thanks for your original PR https://github.com/pytorch/pytorch/pull/114451 as well. It's helpful for us when implementing this support.
